python/utils/LinkedLibraryHolder.cpp

```diff
@@ -30,6 +30,30 @@ constexpr static const char NVQIR_SIMULATION_BACKEND[] =
     "NVQIR_SIMULATION_BACKEND=";
 constexpr static const char TARGET_DESCRIPTION[] = "TARGET_DESCRIPTION=";
 
+int countGPUs() {
+  char buffer[1024];
+  std::string output;
+  FILE *fp1, *fp2;
+
+  fp1 = popen("nvidia-smi", "r");
+  if (NULL == fp1) {
+    cudaq::info("nvidia-smi: command not found");
+    return -1;
+  }
+  pclose(fp1);
+
+  fp2 = popen("nvidia-smi -L | wc -l", "r");
+  if (NULL == fp2) {
+    cudaq::info("nvidia-smi: command not working");
+    return -1;
+  }
+  while (fgets(buffer, sizeof buffer, fp2) != NULL) {
```

Comment:
There is a C++ way of reading in lines from a text line using `std::string`. See the fixup-linkage.cpp file under tools for an example. It's a good way to get around really long lines and not have to deal with the error conditions, etc.

Reply:
Thanks for the pointer to example code. The way I see it to use `getline`, I will have to convert the POSIX file pointer to `ifstream`, which isn't straightforward.